src/tunacode/exceptions.py

```
@@ -304,6 +304,18 @@ def __init__(
         )
 
 
+class ToolRetryError(TunaCodeError):
+    """Raised when a tool needs to signal the agent to retry with a hint.
+
+    This exception is caught by the tool decorator and converted to the
+    framework-specific retry exception (e.g., pydantic_ai.ModelRetry).
+    """
+
+    def __init__(self, message: str, original_error: OriginalError = None):
```

Ruff (0.14.14)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)